roles/hostedbm/tasks/post-deploy-step-on-guest-cluster.yml

```diff
@@ -0,0 +1,115 @@
+---
+- name: "deploy-operators : Install Metallb Operator on the Guest Cluster"
+  ansible.builtin.include_role:
+    name: olm_operator
+    apply:
+      environment:
+        KUBECONFIG: "{{ hostedbm_out_dir }}/{{ hostedbm_cluster_name }}-kubeadmin-kubeconfig"
+  vars:
+    channel: stable
+    install_approval: Automatic
+    operator: metallb-operator
+    namespace: metallb-system
+    source: redhat-operators
+    source_ns: openshift-marketplace
+
+- name: "Set the MetalLB Instance on the Guest Cluster"
```

Comment:
The readme says that the metallb must be installed and configured. Can this be optional or what will be the behavior if the MLB is already configured?

There is already a role in the collection to configure metallb with support for L2 and BGP mode, do you think that can be reused here? https://github.com/redhatci/ansible-collection-redhatci-ocp/tree/main/roles/metallb_setup

Reply:
@betoredhat 


>> The readme says that the metallb must be installed and configured. Can this be optional or what will be the behavior if the MLB is already configured?

We need to install the MetalLB operator on both the hosting and guest clusters. Before executing this role, we assume that you have already configured MetalLB on the hosting cluster (we reuse the `metallb_setup` role for installing on the hosting cluster). In the `post-deploy-step-on-guest-cluster.yml` task, we install MetalLB on the guest cluster.

>> There is already a role in the collection to configure MetalLB with support for L2 and BGP modes. Do you think that role can be reused here? [metallb_setup role](https://github.com/redhatci/ansible-collection-redhatci-ocp/tree/main/roles/metallb_setup)

For the guest cluster, we need to create an `IPAddressPool` with the following additional spec, which this role currently doesn’t support:

```
        protocol: layer2
        autoAssign: false
```

Reply:
We can move this to usage of `metallb_setup role` once that role support above parameters for IPAddresspool

Reply:
@Sandeepyadav93 Understood. Is the installation of MetalLB required on the guest cluster? If not, could we make all the post tasks optional using a flag? Ideally, the default action of the role should focus on the guest cluster installation.

Yes, we need to work on the metallb_setup role to support those specs.

Reply:
@betoredhat MetalLB is required on the guest cluster, as some operators such as console and ingress will not enter a healthy state without it.

Reply:
ok, that sounds good. We haven't had that requirement in our tests with the kubevirt provisioner. But should be good. Let me work on a change to be able to test this role in our lab. Cheers